### PR TITLE
Add example with Ethers-EXT for Klaytn 

### PR DESCRIPTION
**Pull request by nohkwak**
- FeeDelegation example added 
- Add dependency @Klaytn/ethers-ext

### CODE REVIEW

**Review by ian0371 — CHANGES_REQUESTED** [review]
Please also add "package-lock.json"

**Comment by kjeom** [comment]
@nohkwak When I tested, the gasLimit is set by float number.
Please check it again. I think it needs to be casted to int in ethers-ext

```shell
Using contract at: 0xa9eF4a5BfB21e92C06da23Ed79294DaB11F5A6df
number before increment: BigNumber { value: "201" }
{
  type: 49,
  to: '0xa9eF4a5BfB21e92C06da23Ed79294DaB11F5A6df',
  value: 0,
  gasPrice: '0xba43b7400',
  gasLimit: 40084.5,
  from: '0xA2a8854b1802D8Cd5De631E690817c253d6a9153',
  nonce: 213,
  chainId: 1001,
  input: '0xd09de08a'
}
Error: underflow [ See: https://links.ethers.org/v5-errors-NUMERIC_FAULT-underflow ] (fault="underflow", operation="BigNumber.from", value=40084.5, code=NUMERIC_FAULT, version=bignumber/5.7.0)
    at Logger.makeError (/Users/contract-template/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/Users/contract-template/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at throwFault (/Users/contract-template/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts:356:19)
    at Function.BigNumber.from (/Users/contract-template/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts:243:17)
    at FieldTypeNumberBits.canonicalize (/Users/contract-template/node_modules/@klaytn/ethers-ext/src/core/field.ts:190:26)
    at /Users/contract-template/node_modules/@klaytn/ethers-ext/src/core/field.ts:273:39
    at /Users/contract-template/node_modules/lodash/lodash.js:4967:15
    at baseForOwn (/Users/contract-template/node_modules/lodash/lodash.js:3032:24)
    at Function.forOwn (/Users/contract-template/node_modules/lodash/lodash.js:13082:24)
    at TxTypeFeeDelegatedSmartContractExecution.setFields (/Users/contract-template/node_modules/@klaytn/ethers-ext/src/core/field.ts:269:7) {
  reason: 'underflow',
  code: 'NUMERIC_FAULT',
  fault: 'underflow',
  operation: 'BigNumber.from',
  value: 40084.5
}
```

**Comment by nohkwak** [comment]
@kjeom 
The underflow error that you advised above is reflected in https://github.com/klaytn/web3klaytn/pull/418. 
Thanks.

File: script/counter_inc_feePayer.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { deployments, ethers, config, getChainId } from "hardhat";
+import { Wallet as KlaytnWallet } from "@klaytn/ethers-ext";
+import { normalizeHardhatNetworkAccountsConfig } from "hardhat/internal/core/providers/util";
+
+async function getKlaytnSigners(): KlaytnWallet[] {
```
Comment (ian0371) [inline script/counter_inc_feePayer.ts]: How about moving this function to [hardhat-utils](https://github.com/klaytn/hardhat-utils) in the future PR?? If so, should we create a `util` directory there?  @kjeom @blukat29 

We'll just merge this PR for now, and remove this function in the future PR after hardhat-utils supports it.

File: script/counter_inc_feePayer.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  // sender
+  const senderTxHashRLP = await sender.signTransaction(populatedTx);
+  console.log("senderTxHashRLP", senderTxHashRLP);
+
+  // feePayer
+  const decodedTx = feePayer.decodeTxFromRLP(senderTxHashRLP);
+  decodedTx.feePayer = await feePayer.getAddress();
+  console.log(decodedTx);
+
+  const sentTx = await feePayer.sendTransactionAsFeePayer(decodedTx);
```
Comment (ian0371) [inline script/counter_inc_feePayer.ts:44]: Let's remove three console logs:
```
  console.log(decodedTx);
  console.log("sentTx", sentTx);
  console.log("receipt", rc);
```

and add one:
```
  console.log("sending FeePayer transaction", "sender", sender.address, "feePayer", feePayer.address);
```

Comment (nohkwak) [reply script/counter_inc_feePayer.ts:44]: Yes, I updated

File: script/counter_inc_feePayer.ts (hunk 1 above)
Comment (blukat29) [reply script/counter_inc_feePayer.ts]: Oh, hardhat-utils is the place! I've been puzzled to find where to put this.
I strongly agree with your idea.

Comment (ian0371) [reply script/counter_inc_feePayer.ts]: hardhat-utils v0.0.7 has been released. Please update like https://github.com/klaytn/hardhat-utils/pull/10#issuecomment-1628583304 @nohkwak

Comment (nohkwak) [reply script/counter_inc_feePayer.ts]: Yes, I updated. (https://github.com/klaytn/contract-template/pull/1/commits/cc15f51f936dfd085bf07c13bb0fff3da9997fa9)

**Comment by blukat29** [comment]
After https://github.com/klaytn/hardhat-utils/pull/12 is merged and published,
bump hardhat-utils to 0.0.8, and then retry, please.

File: hardhat.config.ts [hunk 3]
```diff
@@ -3,10 +3,10 @@ import "hardhat-deploy";
 import "@nomicfoundation/hardhat-toolbox";
 import "@klaytn/hardhat-utils";
 import "@primitivefi/hardhat-dodoc";
-import * as dotenv from "dotenv";
 
 // the first key of test-junk
-const defaultKey = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+const defaultKey = "0x0e4ca6d38096ad99324de0dde108587e5d7c600165ae4cd6c2462c597458c2b8";
+const defaultKey2 = "0x9435261ed483b6efa3886d6ad9f64c12078a0e28d8d80715c773e16fc000cff4";
```
Comment (ian0371) [inline hardhat.config.ts]: please use private keys from `npx hardhat node`

File: package.json [hunk 4]
```diff
@@ -39,7 +39,8 @@
     "README.md"
   ],
   "devDependencies": {
-    "@klaytn/hardhat-utils": "^0.0.5",
+    "@klaytn/ethers-ext": "^0.9.0-beta",
+    "@klaytn/hardhat-utils": "^0.0.7",
```
Comment (ian0371) [inline package.json]: hardhat-utils should be 0.0.8

Comment (nohkwak) [reply package.json]: Thanks, updated.

File: hardhat.config.ts (hunk 3 above)
Comment (nohkwak) [reply hardhat.config.ts]: Thanks, updated.

File: script/counter_inc_feePayer.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  const feePayer = new KlaytnWallet(accounts[1].privateKey, provider);
+
+  const fields = await counter.populateTransaction.increment();
+  // {
+  //   data: '0xd09de08a',
+  //   to: '0xa9eF4a5BfB21e92C06da23Ed79294DaB11F5A6df',
+  // }
+
+  const tx = {
+    type: 0x31,
```
Comment (blukat29) [inline script/counter_inc_feePayer.ts:24]: TxType.FeeDelegatedSmartContractExecution

File: script/counter_inc_feePayer.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  console.log("senderTxHashRLP", senderTxHashRLP);
+
+  // feePayer
+  const decodedTx = feePayer.decodeTxFromRLP(senderTxHashRLP);
+  console.log(decodedTx);
+  decodedTx.feePayer = feePayer.address;
+
+  const sentTx = await feePayer.sendTransactionAsFeePayer(decodedTx);
+  await sentTx.wait();
+  console.log("sending FeePayer transaction", "sender", sender.address, "feePayer", feePayer.address);
```
Comment (blukat29) [inline script/counter_inc_feePayer.ts]: Fee delegated transaction

**Review by blukat29 — APPROVED** [review]
LGTM except minor comments

File: script/counter_inc_feePayer.ts (hunk 6 above)
Comment (nohkwak) [reply script/counter_inc_feePayer.ts]: Yes, I updated.

File: script/counter_inc_feePayer.ts (hunk 5 above)
Comment (nohkwak) [reply script/counter_inc_feePayer.ts:24]: This can be reflected after ethers-ext 0.9.2-beta is released.

**Review by ian0371 — APPROVED** [review]

**Pull request merged**